sw/device/lib/testing/json/command.h

```diff
@@ -20,6 +20,9 @@ extern "C" {
     value(_, I2cReadTransaction) \
     value(_, I2cWriteTransaction) \
     value(_, I2cWriteTransactionSlow) \
+    value(_, I2cInitInstance0) \
```

Comment:
@a-will, is there a way to send an argument and avoid these additional commands?

Reply:
So this code is just for sending an enum, a first value that can tell the device code what to do and/or listen for next. You can always follow up with another struct, for example, to make a command with arguments.

For example, you'll see I2cTargetAddress here, but the full command request is actually sending a `test_command_t` set to `kTestCommandI2cTargetAddress`, followed by an `i2c_target_address_t`. See the defintion here: https://github.com/lowRISC/opentitan/blob/f87eb7efd2f161278f458676279906c0288a339b/sw/device/lib/testing/json/i2c_target.h#L15-L20

Then there is implementing code, where the test command is used for dispatch. https://github.com/lowRISC/opentitan/blob/f87eb7efd2f161278f458676279906c0288a339b/sw/device/tests/i2c_target_test.c#L236-L238

And then the device knows to pick up the argument. https://github.com/lowRISC/opentitan/blob/f87eb7efd2f161278f458676279906c0288a339b/sw/device/tests/i2c_target_test.c#L80-L82

Reply:
Thanks for pointing me to the right direction, I've added a new generic command that allow the rust code to choose which peripheral instance will be tested.